mozillavpn_en.xlf

```diff
@@ -71,50 +71,56 @@
       <trans-unit id="_msg11">
         <source xml:space="preserve">Mozilla VPN connected</source>
         <target xml:space="preserve"></target>
-        <context-group purpose="location"><context context-type="linenumber">28</context></context-group>
+        <context-group purpose="location"><context context-type="linenumber">34</context></context-group>
       </trans-unit>
       <trans-unit id="_msg12">
         <source xml:space="preserve">TODO</source>
         <target xml:space="preserve"></target>
-        <context-group purpose="location"><context context-type="linenumber">28</context></context-group>
-        <context-group purpose="location"><context context-type="linenumber">32</context></context-group>
-        <context-group purpose="location"><context context-type="linenumber">36</context></context-group>
+        <context-group purpose="location"><context context-type="linenumber">21</context></context-group>
+        <context-group purpose="location"><context context-type="linenumber">34</context></context-group>
+        <context-group purpose="location"><context context-type="linenumber">38</context></context-group>
+        <context-group purpose="location"><context context-type="linenumber">42</context></context-group>
       </trans-unit>
       <trans-unit id="_msg13">
-        <source xml:space="preserve">Mozilla VPN disconnected</source>
+        <source xml:space="preserve">Captive portal detected!</source>
         <target xml:space="preserve"></target>
-        <context-group purpose="location"><context context-type="linenumber">32</context></context-group>
+        <context-group purpose="location"><context context-type="linenumber">21</context></context-group>
       </trans-unit>
       <trans-unit id="_msg14">
+        <source xml:space="preserve">Mozilla VPN disconnected</source>
+        <target xml:space="preserve"></target>
+        <context-group purpose="location"><context context-type="linenumber">38</context></context-group>
+      </trans-unit>
+      <trans-unit id="_msg15">
         <source xml:space="preserve">Mozilla VPN switching</source>
         <target xml:space="preserve"></target>
-        <context-group purpose="location"><context context-type="linenumber">36</context></context-group>
+        <context-group purpose="location"><context context-type="linenumber">42</context></context-group>
       </trans-unit>
     </group>
   </body></file>
   <file original="../src/components/VPNAlert.qml" datatype="plaintext" source-language="en" target-language="en-US"><body>
     <group restype="x-trolltech-linguist-context" resname="VPNAlert">
-      <trans-unit id="_msg15">
+      <trans-unit id="_msg16">
```

Comment:
A few notes:
* The string ID shouldn't change if the string doesn't change. If all strings change when you add a new string to the file, we have a big problem.
* What happens if the translated file doesn't preserve the `<context-group>` part?
* The string with encoded HTML entities is quite terrible to localize. Is that in the original code, or a by-product of the export?

Reply:
>The string ID shouldn't change if the string doesn't change. If all strings change when you add a new string to the file, we have a big problem.
- This seems to be caused by the fact that we use qsTr("text") instead of  qsTrId("id-of-text"); this causes that qt needs to count id's and the 2nd keeps id's consistent. That definitely something we should address asap. 
-Dropping context group should not be a problem, if the trans-unit id's are kept stable. 
- 3rd .... there is html in the source string, so not caused by the export script.

Reply:
OK, that's good news.

> 3rd .... there is html in the source string, so not caused by the export script.

But, it's the HTML written as `Authentication error. &lt;u&gt;&lt;b&gt;Try again&lt;/b&gt;&lt;/u&gt` or `Authentication error. <u><b>Try again</b></u>`?

One more thing: can you add comments to the strings, so that they can be exported as `<notes>` in the XLIFF? If not, we might need to hack something out, e.g. store notes somewhere else (a simple dictionary ID:NOTE), then inject them in the XLIFF XLIFF after each export.

Reply:
Orignal string was: `Authentication error. <u><b>Try again</b></u>` :/ 

the ltool can export an extra note to <extracomment> and <extra-Context> but i could hack it to convert that into <note> later if needed.